StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFlowController.swift

```diff
@@ -591,9 +590,36 @@ extension PaymentSheet {
                 loadResult: self.viewController.loadResult,
                 analyticsHelper: analyticsHelper,
                 walletButtonsShownExternally: self.walletButtonsShownExternally,
-                previousPaymentOption: self._paymentOption
+                previousPaymentOption: self.internalPaymentOption
             )
             self.viewController.flowControllerDelegate = self
+            updatePaymentOption()
+        }
+
+        /// Updates the published paymentOption property based on the current state
+        internal func updatePaymentOption() {
+            if Thread.isMainThread {
```

Comment:
I'm a little worried that this could be quirky depending on all the different things that update this, but I *think* it's okay. (Or at least it's just as reasonable as the old implementation)